components/identities/actions/identityActions.tsx

```diff
@@ -66,45 +72,35 @@ const IdentityActions: FunctionComponent<IdentityActionsProps> = ({
 
   const nextAutoRenew = useMemo(() => {
     const now = Math.floor(Date.now() / 1000);
-    if (identity?.domain_expiry) {
-      if (identity?.domain_expiry + 2592000 < now) {
+    if (identity?.domainExpiry) {
+      if (identity?.domainExpiry + 2592000 < now) {
         return "Next today";
       } else {
         return (
-          "Next on " +
-          timestampToReadableDate(identity?.domain_expiry - 2592000)
+          "Next on " + timestampToReadableDate(identity?.domainExpiry - 2592000)
         );
       }
     }
-  }, [identity?.domain_expiry]);
+  }, [identity]);
 
   // Add all subdomains to the parameters
   const callDataEncodedDomain: (number | string)[] = [encodedDomains.length];
   encodedDomains.forEach((domain) => {
     callDataEncodedDomain.push(domain.toString(10));
   });
 
-  //Set as main domain execute
-  const set_address_to_domain_calls = {
-    contractAddress: process.env.NEXT_PUBLIC_NAMING_CONTRACT as string,
-    entrypoint: "set_address_to_domain",
-    calldata: callDataEncodedDomain,
-  };
-  const set_domain_to_address_calls = {
-    contractAddress: process.env.NEXT_PUBLIC_NAMING_CONTRACT as string,
-    entrypoint: "set_domain_to_address",
-    calldata: [...callDataEncodedDomain, hexToDecimal(address)],
-  };
-  const { writeAsync: set_address_to_domain, data: mainDomainData } =
-    useContractWrite({
-      calls: isAccountTargetAddress
-        ? [set_address_to_domain_calls]
-        : [set_domain_to_address_calls, set_address_to_domain_calls],
-    });
+  const { writeAsync: setMainId, data: mainDomainData } = useContractWrite({
+    calls: identity ? setAsMainId(identity, hasRev, callDataEncodedDomain) : [],
+  });
 
-  function setAddressToDomain(): void {
-    set_address_to_domain();
-  }
+  useEffect(() => {
+    if (!address) return;
+    fetch(`${process.env.NEXT_PUBLIC_SERVER_LINK}/addr_has_rev?addr=${address}`)
+      .then((response) => response.json())
+      .then((jsoned) => {
```

Comment:
`jsoned` ? 

![image](https://github.com/starknet-id/app.starknet.id/assets/78437165/23b3c67b-2fdb-47d4-a9ee-f84ac41a521b)

Reply:
I have to admit I could have been more creative here, here is an improvement: https://github.com/starknet-id/app.starknet.id/pull/580/commits/14d1189c1f7118cf6d2cd6c7152bbacd73bb09a0